src/AppShed/Extensions/StorageBundle/Controller/ApiController.php

```diff
@@ -0,0 +1,444 @@
+        //GET FILTERED DATA
+        $storeData = $this->getDoctrine()->getManager()->getRepository('AppShedExtensionsStorageBundle:Data')->getDataForApi($api, $additionalFilters);
+
+        if ($api->getLimit()) {
+            $storeData = $this->limitResults($storeData, $api->getLimit());
+        }
+        $updateData = $request->request->all();
+
+        $executed = 0;
+        if (! empty($updateData)) {
+            foreach ($storeData as $k => $value) {
+                $data = $storeData[$k]->getData();
+                $newData = array_merge($data, $updateData);
+                $storeData[$k]->setData($newData);
+                $storeData[$k]->setColumns(array_keys($newData));
+            }
+
+            $store = $api->getStore();
+            $em = $this->getDoctrine()->getManager();
+            //Add any new columns to the store
+            $newColumns = array_diff(array_keys($updateData), $storeColumns);
+            if (count($newColumns)) {
+                $store->setColumns(array_merge($storeColumns, $newColumns));
+            }
+            $em->merge($store);
+            $em->flush();
+            $executed = 1;
+        }
+        return ['value' => $executed];
+    }
+
+    private function insertData(Api $api, Request $request) {
+        //get this data here (before getDataForApi()) because $em->clear() make bad
+        $storeColumns = $api->getStore()->getColumns();
+        $appId = $api->getApp()->getAppId();
+
+        $data = $request->request->all();
+        $executed = 0;
+        if (! empty($data)) {
+            $dataO = new Data();
+            $dataO->setStore($api->getStore());
+            $dataO->setColumns(array_keys($data));
+            $dataO->setData($data);
+            $this->getDoctrine()->getManager()->persist($dataO);
+            $this->getDoctrine()->getManager()->flush();
+
+            $store = $api->getStore();
+            $em = $this->getDoctrine()->getManager();
+            //Add any new columns to the store
+            $newColumns = array_diff(array_keys($data), $storeColumns);
+            if (count($newColumns)) {
+                $store->setColumns(array_merge($storeColumns, $newColumns));
+            }
+            $em->merge($store);
+            $em->flush();
+            $executed = 1;
+
+        }
+        return ['value' => $executed];
+    }
+
+    private function selectData(Api $api, Request $request) {
+        $result = [];
+
+        $filters = $request->request->get('filters', '');
+        $additionalFilters = $this->getAdditionalFilters($filters);
+        //GET FILTERED DATA
+        $storeData = $this->getDoctrine()->getManager()->getRepository('AppShedExtensionsStorageBundle:Data')->getDataForApi($api, $additionalFilters);
+
+        $sql['select'] = [];
+        foreach ($api->getFields() as $field) {
+            $sql['select'][] = $field->getField();
+            if ($field->getAggregate()) {
```

Comment:
You could have made aggregateField a separate property of `Api` and then you wouldn't have to do this!